lib/galaxy/config/__init__.py

```diff
@@ -620,8 +620,36 @@ class GalaxyAppConfiguration(BaseAppConfiguration, CommonConfigurationMixin):
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self._override_tempdir(kwargs)
+        self._configure_sqlalchemy20_warnings(kwargs)
         self._process_config(kwargs)
 
+    def _configure_sqlalchemy20_warnings(self, kwargs):
+        """
+        This method should be deleted after migration to SQLAlchemy 2.0 is complete.
+        To enable warnings, set `GALAXY_CONFIG_SQLALCHEMY_WARN_20=1`,
+        """
+        warn = kwargs.get('sqlalchemy_warn_20', False)
+        if warn:
+            import sqlalchemy
+            sqlalchemy.util.deprecations.SQLALCHEMY_WARN_20 = True
+            self._setup_sqlalchemy20_warnings_filters()
+
+    def _setup_sqlalchemy20_warnings_filters(self):
+        import warnings
+        from sqlalchemy.exc import RemovedIn20Warning
+        # Always display RemovedIn20Warning warnings.
+        warnings.filterwarnings('always', category=RemovedIn20Warning)
```

Comment:
Is there a way to elevate these to error ? otherwise we have to change the log level for sentry.

Reply:
I can't see anything obvious in the warnings module documentation on this. I'm sure I can come up with a solution, but I suspect that will involve intercepting all warnings and logging an error for those that match some criteria. I may be wrong though and there may be a simpler solution. But if we can just change the log level for sentry, I think that would be a better solution.

Reply:
Yeah, we can do that.